lib/Sema/CSDiagnostics.cpp

```diff
@@ -3399,6 +3402,30 @@ bool MissingMemberFailure::diagnoseForDynamicCallable() const {
   return false;
 }
 
+bool MissingMemberFailure::diagnoseForDefaultAnyArrayLiteral() const {
+  auto &cs = getConstraintSystem();
+  auto *expr = getAsExpr(getAnchor());
+  auto *parentExpr = cs.getParentExpr(expr);
+  auto contextualType = getContextualType(parentExpr);
+  auto baseType = resolveType(getBaseType())->getWithoutSpecifierType();
+
+  if (contextualType)
+    return false;
```

Comment:
Why should this matter?

Reply:
For cases like, `let _ : [Any] = [.e]`,  if there is a contextual type, I thought that the old diagnostics makes more sense.